erts/emulator/beam/erl_gc.h

```diff
@@ -193,6 +193,13 @@ int erts_dbg_within_proc(Eterm *ptr, Process *p, Eterm* real_htop);
 /* Validates the frame chain, ensuring that it always points within the stack
  * and that no frames are skipped. */
 void erts_validate_stack(Process *p, Eterm *frame_ptr, Eterm *stack_top);
+int
+erts_dbg_check_heap_terms(int (*check_eterm)(Eterm),
+                          Process *p,
+                          Eterm *real_htop);
```

Comment:
If we always abort on failure, why return `int`?

Reply:
It is to make the debug function more general and useful for other things one might want to use it for in the future. If the check_eterm function provided as an argument return 0 for any of the term on the heap, the whole function returns 0 instead of 1. This could be useful, for example, if one wants to log something when a particular term is found instead of aborting.

Reply:
I think it would be better to add that when we need it, but I don't have a strong opinion and leave it up to you. It's a small nit anyway :)